### PR TITLE
JP-3547: Fix MIRI array sizing for multiple inputs and pixel ratio != 1

### PR DESCRIPTION
**Pull request by melanieclarke**
Resolves [JP-3547](https://jira.stsci.edu/browse/JP-3547)

Fixes an issue discovered during testing for #8596.

For MIRI, when there are multiple input files and pixel_scale_ratio is set to something other than 1.0, the output array size was being computed incorrectly: the pixel_scale_ratio was applied twice.  This PR fixes that issue and adds a unit test for the multiple input case.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] ~updated relevant documentation~
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by melanieclarke** [comment]
Regression tests here:
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1677

**Comment by melanieclarke** [comment]
Regtest failures all look unrelated - there are no changes to any spectral resampling products.

File: jwst/resample/tests/test_resample_step.py
Comment (emolter) [inline jwst/resample/tests/test_resample_step.py:501]: what is the purpose of setting the filenames?

File: jwst/resample/tests/test_resample_step.py
Comment (melanieclarke) [reply jwst/resample/tests/test_resample_step.py:515]: Not quite - they are close, but not necessarily exactly identical at the edges because of floating point precision.  For this test, when pixel ratio is 0.7, there are 2 pixels that end up NaN when pixel_scale is specified that are not NaN when the equivalent pixel_scale_ratio is specified.

Comment (emolter) [reply jwst/resample/tests/test_resample_step.py:515]: sounds good, as long as it's just edge effects

**Comment by melanieclarke** [comment]
@tapastro - can you please review for the maintainers?

**Review by tapastro — APPROVED** [review]
LGTM - will wait for CI then merge.

**Pull request merged**
